### ISSUE TITLE
remote error: tls: unexpected message

### ISSUE THREAD

**Issue by Harusakii** [bogdanfinn/tls-client#6]
It seems to work at pretty much every site, but when you make a simple GET request to google.com, it receives the error `remote error: tls: unexpected message`

It goes off in the utls package at line 698 [[link to code]](https://github.com/bogdanfinn/utls/blob/9f832c54e974dbff1c57e4b4e7a0c3ab83bb671f/conn.go#L698)

Using the normal Chrome 105 client, Header and Headerorder perfectly set.

Thanks for helping

**Comment by 89z** [comment]
I have an alternate package that works:

~~~go
package main

import (
   "github.com/89z/rosso/crypto"
   "net/http"
)

func main() {
   req, err := http.NewRequest("GET", "https://www.google.com", nil)
   if err != nil {
      panic(err)
   }
   spec, err := crypto.Parse_JA3(crypto.Android_API_26)
   if err != nil {
      panic(err)
   }
   res, err := crypto.Transport(spec).RoundTrip(req)
   if err != nil {
      panic(err)
   }
   defer res.Body.Close()
   println(res.StatusCode == 200)
}
~~~

**Comment by Harusakii** [comment]
I found out the problem, it was the generic extension in the ClientSpec that made problems: 
&tls.GenericExtension {id: 0x4469}

it brickes some sites

**Comment by bogdanfinn** [comment]
@Harusakii this issue was reported already several times to me that for some reason the predefined chrome clients do not work on the normal google.com page.

In my implementation there is no `GenericExtension` in use anymore because i replaced them by the actualy implementations made in the community. Nevertheless for chrome profiles (103, 104, 105) the extension which breaks the google.com page is `&ALPSExtension{SupportedProtocols: []string{"h2"}},` 

Yeah you could argue removing that extension but then the fingerprint would not be chrome anymore because according to wireshark logs this extension is used by chrome.

So just removing the extension is sadly not the solution. Also just using another tls client profile is not the solution as it was suggested in the already deleted comments.

Glad for us that the chrome profile works "at pretty much every site" and google.com is not that important for us.

But this issue is still on my list yeah.

**Comment by Harusakii** [comment]
> @Harusakii this issue was reported already several times to me that for some reason the predefined chrome clients do not work on the normal google.com page.
> 
> In my implementation there is no `GenericExtension` in use anymore because i replaced them by the actualy implementations made in the community. Nevertheless for chrome profiles (103, 104, 105) the extension which breaks the google.com page is `&ALPSExtension{SupportedProtocols: []string{"h2"}},`
> 
> Yeah you could argue removing that extension but then the fingerprint would not be chrome anymore because according to wireshark logs this extension is used by chrome.
> 
> So just removing the extension is sadly not the solution. Also just using another tls client profile is not the solution as it was suggested in the already deleted comments.
> 
> Glad for us that the chrome profile works "at pretty much every site" and google.com is not that important for us.
> 
> But this issue is still on my list yeah.

yeah, tested out and it indeed was ALPS, I think I just mistook the genericextension bc I deleted them both at the same time (to test haha)

and yeah, as long ALPS is used, but doesnt properly work, I guess just switching it off on the sites, where it doesnt work is the way to go now rn

**Comment by bogdanfinn** [comment]
@Harusakii seems like here is a solution for this issue:

https://github.com/refraction-networking/utls/commit/8d0f1c40c755c9b68f3d8bc926d4934f8ac4286e

**Comment by bogdanfinn** [comment]
@Harusakii 
should be resolved now with [#v0.8.3](https://github.com/bogdanfinn/tls-client/releases/tag/v0.8.3)

**Comment by Harusakii** [comment]
> @Harusakii should be resolved now with [#v0.8.3](https://github.com/bogdanfinn/tls-client/releases/tag/v0.8.3)

tried out and it didnt :/



> @Harusakii seems like here is a solution for this issue:
> 
> [refraction-networking/utls@8d0f1c4](https://github.com/refraction-networking/utls/commit/8d0f1c40c755c9b68f3d8bc926d4934f8ac4286e)

implemented this and it works! maybe implement this in your [utls](https://github.com/bogdanfinn/utls)?

**Comment by bogdanfinn** [comment]
@Harusakii i implemented it in version 0.8.3

make sure your go.mod has these packages in the following versions:

```
	github.com/bogdanfinn/fhttp v0.5.8 
	github.com/bogdanfinn/tls-client v0.8.3 
	github.com/bogdanfinn/utls v1.5.8 
```

I implemented that change in utls v1.5.8 
https://github.com/bogdanfinn/utls/releases/tag/v1.5.8

**Comment by Harusakii** [comment]
> @Harusakii i implemented it in version 0.8.3
> 
> make sure your go.mod has these packages in the following versions:
> 
> ```
> 	github.com/bogdanfinn/fhttp v0.5.8 
> 	github.com/bogdanfinn/tls-client v0.8.3 
> 	github.com/bogdanfinn/utls v1.5.8 
> ```
> 
> I implemented that change in utls v1.5.8 https://github.com/bogdanfinn/utls/releases/tag/v1.5.8

ohhh I see, I just assumed the master branch in utls was updated thats why haha, thank you very much!
